profiles/drupalru/modules/dru_tnx/includes/dru_tnx.crud.inc

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/**
+ * @file
+ *   Data Base functionality
+ */
+
+/**
+ * Implementation get_user_tnx_statistic().
+ *
+ * Return user statistic object
+ */
+function get_user_tnx_statistic($uid) {
+  $statistic = db_select('tnx_stat', 't')
+    ->fields('t')
+    ->condition('t.uid', $uid)
+    ->execute()
+    ->fetchObject();
+
+  // If user don't have statistic.
+  if (!$statistic) {
+    $statistic = new stdClass();
+    $statistic->uid = $uid;
+    $statistic->tnx_node = 0;
+    $statistic->tnx_comment = 0;
+    $statistic->node = 0;
+    $statistic->comment = 0;
+  }
+
+  return $statistic;
+}
+
+/**
+ * Implementation get_user_tnx().
+ *
+ * Return Boolean TRUE if user already have thanked.
+ *
+ * @param string $entity_type Entity type ("node" or "comment").
+ * @param int    $entity_id   Entity ID (NID or CID).
+ * @param int    $uid         Thanked user ID (UID).
+ *
+ * @return bool
+ */
+function get_user_tnx($entity_type, $entity_id, $uid) {
+  $tnx = db_select('tnx', 't')
+    ->fields('t', ['id'])
+    ->condition('t.entity_type', $entity_type)
+    ->condition('t.entity_id', $entity_id)
+    ->condition('t.uid', $uid)
+    ->execute()
+    ->fetchAll();
+
+  return !$tnx;
```

Comment:
If you need just check an existing of records, then use `->countQuery()`, it's will be better for performance.